src/chunk.c

```diff
@@ -849,6 +925,43 @@ append_chunk_oid(ChunkScanCtx *scanctx, Chunk *chunk)
 	return true;
 }
 
+static bool
+is_numeric_type(Oid type)
+{
+	return type == INT8OID || type == INT4OID || type == INT2OID;
+}
+
+static bool
+is_valid_time_column_type(Oid type)
+{
+	return is_numeric_type(type) ||type == InvalidOid || type == DATEOID ||
+	type == TIMESTAMPOID || type == TIMESTAMPTZOID;
+}
+static void
```

Comment:
Can you provide a comment with the logic for this check and the reasoning behind the logic